src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -363,7 +363,7 @@ private int getOutputSizeByOperation(int inputLen, boolean isDoFinal) {
         case GCM_MODE:
             if (isDoFinal) {
                 int tagLen = 0;
-                 if (useNativeGCM) {
+                 if (useNativeGCM && NativeCrypto.isAllowedAndLoaded()) {
                     tagLen = ((NativeGaloisCounterMode) cipher).getTagLen();
                 } else {
                     tagLen = ((GaloisCounterMode) cipher).getTagLen();
```

Comment:
This test is needlessly expensive; we only need to decide which implementation of `getTagLen()` to call:
```java
                int tagLen;
                if (cipher instanceof NativeGaloisCounterMode) {
                    tagLen = ((NativeGaloisCounterMode) cipher).getTagLen();
                } else {
                    tagLen = ((GaloisCounterMode) cipher).getTagLen();
                }
```

Reply:
Fixed.